### ISSUE TITLE
test-keys/end.pem isn't shipped as part of ntp-proto

### ISSUE THREAD

**Issue by sylvestre** [pendulum-project/ntpd-rs#1332]
Just like:
https://github.com/pendulum-project/ntpd-rs/issues/1214

ntp-proto tests are failing with:
```
280s error: couldn't read src/../../test-keys/end.pem: No such file or directory (os error 2)
280s     --> src/nts_record.rs:2897:42
280s      |
280s 2897 |             &mut std::io::BufReader::new(include_bytes!("../../test-keys/end.pem") as &[u8]),
280s      |                                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
280s      |
280s      = note: this error originates in the macro `include_bytes` (in Nightly builds, run with -Z macro-backtrace for more info)
280s 
282s error: could not compile `ntp-proto` due to previous error
282s 
282s Caused by:
282s   process didn't exit successfully: `CARGO=/usr/bin/cargo CARGO_CRATE_NAME=ntp_proto CARGO_MANIFEST_DIR=/usr/share/

```
full log:
https://ci.debian.net/packages/r/rust-ntp-proto/testing/amd64/42362802/

**Comment by rnijveld** [comment]
That's a shame. I think we should change the directory structure such that at least this specific case cannot happen anymore. My proposal is that we remove the top level test-keys folder in the project and move it to either ntp-proto or ntpd, and then we duplicate/symlink it to the other crate. Additionally, maybe we want to add a CI step that attempts to test the crates isolated from the rest of the repository before we create a release, making sure things like this don't happen anymore. Would also be nice if `cargo publish` would support a `--test` flag for this, it currently only does a build of the crate itself, it doesn't build nor run the test code.

**Comment by sylvestre** [comment]
Yeah, duplicating isn't a big deal :)

**Comment by davidv1992** [comment]
Fixed the immediate issue again in version 1.1.2. Leaving it up to @rnijveld if this is a sufficient solution for now or if we want to do more before closing this.

**Comment by rnijveld** [comment]
I've created a separate issue for that, see #1337
